### ISSUE TITLE
Regression: DLL loading problem (Entry Point Not Found)

### ISSUE THREAD

**Issue by MartinDupuisLorez** [sandboxie-plus/Sandboxie#2980]
### Describe what you noticed and did

I'm using a particular version of a program that previously worked fine with Sandboxie and when I last updated to v1.9.5 it crashed as soon as I launched the exe, with the following error:

Title: **Entry Point Not Found**
Message: The procedure entre point [lib] could not be located in the dynamic link librairy [exe path]

------------------

Since this is a regression (since the executable worked without problems before), I have tested all versions of Sandboxie since then, in order to identify the one that introduced the problem, **conclusion: it is version 1.7.0**.

FYI, here are the results of my tests with different versions of Sandboxie (just to be clear, I never modified the configuration of the Sandbox used for this test):
- Sandboxie Plus v1.6.7: it works ✅.
- Sandboxie Plus v1.7.0: it crashes ❌.
- Sandboxie Plus v1.9.5 (latest version available to date): it crashes ❌.

### How often did you encounter it so far?

Systematically since Sandboxie Plus v1.7.0

### Affected program

Antidote.exe

### Download link

Modified version is not publicly available, but could be shared upon request

### Where is the program located?

The program is installed only inside a sandbox (NOT in the real system anyway).

### Expected behavior

The exe should run fine on latest version.

### What is your Windows edition and version?

Windows 10 Enterprise  22H2

### In which Windows account you have this problem?

A local account (Administrator).

### Please mention any installed security software

Microsoft Defender Antivirus

### What version of Sandboxie are you running?

Sandboxie Plus v1.9.5 64-bit

### Is it a new installation of Sandboxie?

I recently did a new clean installation.

### Is it a regression?

Sandboxie Plus v1.7.0

### In which sandbox type you have this problem?

In a standard isolation sandbox (yellow sandbox icon).

### Can you reproduce this problem on a new empty sandbox?

I can confirm it also on a new empty sandbox.

### Crash dump

Not available (crash don’t generate any crash dump)

### Trace log

- [Trace log for Sandboxie Plus v1.6.7](https://upload.disroot.org/r/1qagXF7A#aAfA2oiu726DP8Hky5ZG0je9V5WTZeQWckwRWmc//QQ=) (.exe works ✅)
- [Trace log for Sandboxie Plus v1.7.0](https://upload.disroot.org/r/4_XpEmjt#AtI0BIE1oDgllF5QBcXKrYkS1Gfc5cl1ykh8+bVRl38=) (.exe crashes ❌)

**Comment by DavidXanatos** [comment]
not cool... seams this particular application does not like the new CET compatible entrypoint hook https://github.com/sandboxie-plus/Sandboxie/issues/2559

Not sure how I can fix that properly without having access to the application in question for testing.
Implementing a user switch to revert back to the old behavior is well... quite unelegant.

Would it be possible to provide the problematic software for testing by email?

**Comment by MartinDupuisLorez** [comment]
@DavidXanatos I understand! I just emailed you the application and its environment to be able to easily launch it on your side.

**Comment by DavidXanatos** [comment]
thank you I was able to reproduce the issue and will be soon looking into how to fix it properly.

**Comment by DavidXanatos** [comment]
I have found a fix for the issue, the new CET compatible hook code was 7 bytes longer then the old hacky one seams this is to much for the failing application. Apparently the hook overwrites a part of an other code block which is used when CRT loads all the Qt dll's, that happens after we set the hook but before we actually ran into it and restored the original code.
I have re worked the new code to use a global variable instead of re capturing the entry point address from the instruction pointer.

According to a adjacent comment the reason to re capturing the entry point address was an incompatibility Microsoft User Experience Virtualization observed for 32-bit processes, as the CET enabled code is used only in 64-bit mode it seams safe to go with using the global variable and the simplified hook.
